test/formal/redundant_store_unrelated.py

```diff
@@ -0,0 +1,55 @@
+from z3 import Solver, Int, IntVal, unsat
+import sys
+
+"""
+Tests that the conditions inside RedundantStoreEliminator::knownUnrelated
+properly account for overflow.
+
+"""
+
+n_bits = 256
+
+solver = Solver()
+solver.set("timeout", 60000)
+
+def restrict(x):
+    solver.add(x >= 0)
+    solver.add(x < 2**n_bits)
+
+def restrictedInt(x):
+    var = Int(x)
+    restrict(var)
+    return var
+
+start1 = restrictedInt('start1')
+length1 = restrictedInt('length1')
+start2 = restrictedInt('start2')
+length2 = restrictedInt('length2')
+
+k = Int('k')
+diff = Int('diff')
+solver.add(diff == start2 - start1 + k * 2**n_bits)
```

Comment:
If we go with the assumption that all memory access is "small", this file needs to be updated to show that the condition is correct under that assumption.